src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,232 @@
+         {"CYP2C19", createFactors(0.32, 0.26, 0.12)},
+         {"CYP2D6", createFactors(0.76, 0.33, 0.11)},
+      };
+
+      public const string CHILD_PUGH_SCORE = "Child-Pugh score";
+
+      private const int HI_EDGINTON_VALUE_ORIGIN_ID = 93;
+      private const int HI_JOHNSON_VALUE_ORIGIN_ID = 94;
+
+      private readonly IDimension _ageDimension;
+
+      public HIDiseaseStateImplementation(IDimensionRepository dimensionRepository,
+         IParameterFactory parameterFactory,
+         IValueOriginRepository valueOriginRepository,
+         IFormulaFactory formulaFactory,
+         IIndividualFactory individualFactory,
+         IContainerTask containerTask,
+         IParameterSetUpdater parameterSetUpdater) : base(valueOriginRepository, formulaFactory, individualFactory, containerTask, parameterSetUpdater, CoreConstants.DiseaseStates.HI)
+      {
+         _parameterFactory = parameterFactory;
+         _ageDimension = dimensionRepository.AgeInYears;
+      }
+
+      public override bool ApplyTo(Individual individual) => applyTo(individual, UpdateParameter(HI_EDGINTON_VALUE_ORIGIN_ID));
+
+      public override bool ApplyForPopulationTo(Individual individual) => applyTo(individual, UpdateParameterValue);
+
+      private bool applyTo(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
+      {
+         updateBloodFlows(individual, updateParameterFunc);
+         updateVolumes(individual, updateParameterFunc);
+         updateGFR(individual, updateParameterFunc);
+         updateOntogenyFactory(individual, updateParameterFunc);
+         updateHematocrit(individual, updateParameterFunc);
+
+         addChildPughScoreToLiver(individual);
+         return true;
+      }
+
+      private void addChildPughScoreToLiver(Individual individual)
+      {
+         var organism = individual.Organism;
+         var childPughScore = organism.Parameter(CHILD_PUGH_SCORE);
+         if (childPughScore == null)
+         {
+            childPughScore = createChildPughScore();
+            organism.Add(childPughScore);
+         }
+
+         childPughScore.Value = childPughScoreFor(individual);
+      }
+
+      private IParameter createChildPughScore()
+      {
+         var childPughScore = _parameterFactory.CreateFor(CHILD_PUGH_SCORE, PKSimBuildingBlockType.Individual)
+            .WithGroup(CoreConstants.Groups.INDIVIDUAL_CHARACTERISTICS);
+         childPughScore.DefaultValue = null;
+         childPughScore.Editable = false;
+         return childPughScore;
+      }
+
+      protected override IReadOnlyList<IParameter> ParameterChangedByDiseaseStateAsList(Individual individual)
+      {
+         var organism = individual.Organism;
+         var kidney = organism.Organ(KIDNEY);
+         var liver = organism.Organ(LIVER);
+
+         var organsBloodFlow = new[] {STOMACH, SMALL_INTESTINE, LARGE_INTESTINE, SPLEEN, PANCREAS, LIVER, KIDNEY, BONE, FAT, GONADS, HEART, MUSCLE, SKIN};
```

Comment:
Could you sort the organ list alphabetically?
Also, BRAIN is missing in the list - on purpose? (pls add a comment then)

Reply:
I followed the listas in the paper so that I would not miss one. good point with brain. I will ask